tests/test_cli.py

```
@@ -1486,10 +1500,40 @@ def test_handle_auth_login_non_interactive_mode_success(
+    @patch("mmrelay.cli.ensure_directories")
+    @patch("builtins.print")
+    def test_handle_auth_login_propagates_explicit_config(
+        self, mock_print, mock_ensure_dirs, mock_login
+    ):
+        """Pass the resolved CLI config into login for E2EE/path decisions."""
+        config_data = {"matrix": {"e2ee": {"enabled": True}}}
+        self.mock_load_config_silently.return_value = config_data
+        self.mock_args.config = "/custom/config.yaml"
+        self.mock_args.homeserver = "https://matrix.example"
+        self.mock_args.username = "@bot:matrix.example"
+        self.mock_args.password = "secret123"
```

Ruff (0.15.21)

[error] Possible hardcoded password assigned to: "password"

(S105)